src/modules/pair/services/pair.compute.service.ts

```diff
@@ -654,4 +654,82 @@ export class PairComputeService implements IPairComputeService {
         );
         return deployedAt ?? undefined;
     }
+
+    @ErrorLoggerAsync({
+        logArgs: true,
+    })
+    @GetOrSetCache({
+        baseKey: 'pair',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async pairFarmAddress(pairAddress: string): Promise<string> {
+        return await this.computePairFarmAddress(pairAddress);
+    }
+
+    async computePairFarmAddress(pairAddress: string): Promise<string> {
```

Comment:
move these methods to `pair.service.ts`. No need for caching because all the data you are using is already cached

Reply:
I've renamed methods and removed caching